src/server/scripts/Shadowlands/SanctumOfDomination/boss_sylvanas_windrunner.cpp

```diff
@@ -0,0 +1,429 @@
+    }
+
+    void MovementInform(uint32 type, uint32 id) override
+    {
+        if (type != POINT_MOTION_TYPE)
+            return;
+
+        if (id == POINT_INTRO_01 || id == POINT_INTRO_02 || id == POINT_INTRO_03)
+            DoCastSelf(SPELL_GENERIC_ANCHOR_HERE, true);
+    }
+
+    void KilledUnit(Unit* victim) override
+    {
+        if (!victim->IsPlayer())
+            return;
+
+        // TODO: there should be an internal CD for this.
+        Talk(SAY_SLAY);
+    }
+
+    void JustEngagedWith(Unit* who) override
+    {
+        BossAI::JustEngagedWith(who);
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_ENGAGE, me);
+
+        Talk(SAY_AGGRO);
+
+        instance->DoUpdateWorldState(WORLD_STATE_SYLVANAS_ENCOUNTER_PHASE, PHASE_ONE);
+
+        events.SetPhase(PHASE_ONE);
+
+        // Note: Sylvanas uses her root with 2s at the beginning of the encounter, most likely to avoid moving when engaging at stance switch.
+        DoCastSelf(SPELL_SYLVANAS_ROOT, CastSpellExtraArgs(TRIGGERED_FULL_MASK).AddSpellMod(SPELLVALUE_DURATION, 2000));
+
+        // Note: we won't allow engaging until Phase 1 PR is merged.
+        EnterEvadeMode(CreatureAI::EvadeReason::EVADE_REASON_OTHER);
+    }
+
+    void DoAction(int32 action) override
+    {
+        switch (action)
+        {
+            case ACTION_START_SYLVANAS_INTRO:
+                events.ScheduleEvent(EVENT_INTRODUCTION, 10s, EVENT_GROUP_INTRODUCTION_EVENTS, PHASE_ONE);
+                break;
+            default:
+                break;
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        events.Update(diff);
+
+        if (me->HasUnitState(UNIT_STATE_CASTING))
+            return;
+
+        while (uint32 eventId = events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_INTRODUCTION:
+                    Conversation::CreateConversation(CONVERSATION_INTRO, me, me->GetPosition(), ObjectGuid::Empty);
+                    events.ScheduleEvent(EVENT_INTRODUCTION + 1, 1s + 500ms, EVENT_GROUP_INTRODUCTION_EVENTS);
+                    break;
+
+                case EVENT_INTRODUCTION + 1:
+                    me->SendPlaySpellVisualKit(SPELL_VISUAL_KIT_SYLVANAS_INTRODUCTION_TALK_01, 0, 0);
+                    if (Creature* bolvar = instance->GetCreature(DATA_BOLVAR_FORDRAGON_PINNACLE))
+                        me->SetFacingToObject(bolvar);
+                    events.ScheduleEvent(EVENT_INTRODUCTION + 2, 1s + 140ms, EVENT_GROUP_INTRODUCTION_EVENTS);
+                    break;
+
+                case EVENT_INTRODUCTION + 2:
+                    me->GetMotionMaster()->MovePoint(POINT_INTRO_01, SylvanasIntroPos[0], false);
+                    events.ScheduleEvent(EVENT_INTRODUCTION + 3, 1s + 500ms, EVENT_GROUP_INTRODUCTION_EVENTS);
+                    break;
+
+                case EVENT_INTRODUCTION + 3:
+                    me->SendPlaySpellVisualKit(SPELL_VISUAL_KIT_SYLVANAS_INTRODUCTION_TALK_02, 0, 0);
```

Comment:
i assume this should be executed when she reached POINT_INTRO_01?

Reply:
It should happen mid-way iirc.